src/modules/document-caching/AmpDocumentCachablePlugin.ts

```diff
@@ -28,9 +40,20 @@ export default class AmpDocumentCachablePlugin extends Plugin {
   }): Promise<Response | null> {
     const clonedResponse = response.clone();
     const responseContentType = clonedResponse.headers.get('content-type');
+
     // TODO: implement header check as well as it'll be less work.
     if (responseContentType && responseContentType.includes('text/html')) {
       try {
+        // Check if the url qualifies for any explicitly allowed page.
+        if (this.allowedPages_) {
+          for (let index = 0; index < this.allowedPages_.length; index++) {
+            const currentTestUrl = this.allowedPages_[index];
+            if (currentTestUrl.test(clonedResponse.url)) {
+              return response;
+            }
+          }
```

Comment:
Might also consider an `Array.find`.

Reply:
fixed with `Array.some` as we were not using the returned object